### PR TITLE
Basic support for Metal

### PR DESCRIPTION
**Pull request by kvark**
Addresses part of #24

~~I haven't actually made it running, since the lack of "vulkan.h" on the system. We should be packing it in this project. But at least the code builds now on Metal.~~

Update: PR has been revived. Metal builds and executes just fine up to the point where window integration is needed - leaving for the follow-ups (help is appreciated!).

### CODE REVIEW

**Comment by grovesNL** [comment]
Looks like we'll need to update the compiler (either GCC or Clang), at least for GL. You could probably just copy the Travis configuration from gfx for now unless you want to try to simplify it here.

File: libportability-gfx/src/lib.rs [hunk 1]
```diff
@@ -4,10 +4,14 @@
 #![allow(improper_ctypes)] //TEMP: buggy Rustc FFI analysis
 
 extern crate gfx_hal as hal;
+#[cfg(feature = "metal")]
+extern crate gfx_backend_metal as back;
 #[cfg(feature = "dx12")]
 extern crate gfx_backend_dx12 as back;
 #[cfg(feature = "vulkan")]
 extern crate gfx_backend_vulkan as back;
+#[cfg(not(any(feature = "metal", feature = "dx12", feature = "vulkan")))]
```
Comment (msiglreith) [inline libportability-gfx/src/lib.rs]: Why not just `#[cfg(feature = "metal")]`? Maybe the `empty` backend as default case?

**Comment by kvark** [comment]
@msiglreith please take another look. I had to hide the new dispatch handle behind a feature, also switch a few things to it. Plus, the whole new CI script.

File: libportability-gfx/src/lib.rs [hunk 2]
```diff
@@ -29,8 +31,8 @@ use std::collections::HashMap;
 pub use impls::*;
 
 // Vulkan objects
-pub type VkInstance = Handle<back::Instance>;
-pub type VkPhysicalDevice = Handle<hal::Adapter<B>>;
+pub type VkInstance = DispatchHandle<back::Instance>;
```
Comment (msiglreith) [inline libportability-gfx/src/lib.rs]: Almost missed it but Instance does **not** have the loader field.

**Comment by kvark** [comment]
@msiglreith notes addressed, thanks for the review!

**Comment by msiglreith** [comment]
✈️ bors r+

**Comment by kvark** [comment]
bors r=msiglreith

**Pull request merged**